debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/RecordsSnapshotProducer.java

```diff
@@ -172,37 +190,33 @@ private void takeSnapshot(BlockingConsumer<ChangeEvent> consumer) {
+            Duration lockTimeout = taskContext.config().snapshotLockTimeout();
             logger.info("Step 1: starting transaction and refreshing the DB schemas for database '{}' and user '{}'",
                         connection.database(), connection.username());
-            // we're using the same isolation level that pg_backup uses
-            StringBuilder statements = new StringBuilder("SET TRANSACTION ISOLATION LEVEL SERIALIZABLE, READ ONLY, DEFERRABLE;");
-            connection.executeWithoutCommitting(statements.toString());
-            statements.delete(0, statements.length());
+
+            String transactionStatement = snapshotter.createSnapshotTransaction(slotCreatedInfo);
+            logger.info("openining transaction with statement {}", transactionStatement);
+            connection.executeWithoutCommitting(transactionStatement);
 
             //next refresh the schema which will load all the tables taking the filters into account
             PostgresSchema schema = schema();
             schema.refresh(connection, false);
 
-            logger.info("Step 2: locking each of the database tables, waiting a maximum of '{}' seconds for each lock",
-                        lockTimeoutMillis / 1000d);
-            statements.append("SET lock_timeout = ").append(lockTimeoutMillis).append(";").append(lineSeparator);
-            // we're locking in SHARE UPDATE EXCLUSIVE MODE to avoid concurrent schema changes while we're taking the snapshot
-            // this does not prevent writes to the table, but prevents changes to the table's schema....
-            // DBZ-298 Quoting name in case it has been quoted originally; it doesn't do harm if it hasn't been quoted
-            schema.tableIds().forEach(tableId -> statements.append("LOCK TABLE ")
-                                                         .append(tableId.toDoubleQuotedString())
-                                                         .append(" IN SHARE UPDATE EXCLUSIVE MODE;")
-                                                         .append(lineSeparator));
-            connection.executeWithoutCommitting(statements.toString());
-
-            //now that we have the locks, refresh the schema
-            schema.refresh(connection, false);
+            Optional<String> lockStatement = snapshotter.lockTablesForSnapshot(lockTimeout, schema.tableIds());
+            if (lockStatement.isPresent()) {
+                logger.info("Step 2: locking each of the database tables, waiting a maximum of '{}' seconds for each lock",
+                        lockTimeout.getSeconds());
+                connection.executeWithoutCommitting(lockStatement.get());
+                //now that we have the locks, refresh the schema
+                schema.refresh(connection, false);
+            }
+            else {
+                logger.warn("Step 2: skipping locking each table, this may result in inconsistent schema!");
```

Comment:
Is this actually true when using exported snapshots? It seems that message might make things sound worse than they are in this case?